CONTRIBUTING.md

```diff
@@ -0,0 +1,157 @@
+# Contributing to document-service
+
+Welcome to the document-service project! We appreciate your interest in contributing to the project and making it even better. As a contributor, 
+please follow the guidelines outlined below:
+
+## Table of Contents
+- [Got a Question or Problem?](#got-a-question-or-problem)
+- [Issues and Bugs](#found-any-issues-and-bugs)
+- [Submitting an Issue](#submission-guidelines)
+- [Feature Requests](#feature-requests)
+- [Coding Rules](#coding-rules)
+- [Commit Message Guidelines](#commit-message-guidelines)
+
+## Got a Question or Problem?
+
+**If you have questions or encounter problems, please refrain from opening issues for general support questions**. GitHub issues are primarily for bug 
+reports and feature requests. For general questions and support, consider using [Stack Overflow](https://stackoverflow.com/questions/tagged/document-service) 
+and tag your questions with the `document-service` tag. Here's why Stack Overflow is a preferred platform:
+
+	- Questions and answers are publicly available, helping others.
+	- The voting system on Stack Overflow highlights the best answers.
+
+To save time for both you and us, we will close issues related to general support questions and direct users to Stack Overflow.
+
+## Found any Issues and Bugs
+
+If you find a bug in the source code, you can help us by [submitting an issue](https://github.com/OsmosysSoftware/document-service/issues/new) 
+to our GitHub Repository. Even better, you can submit a (Pull Request)[https://github.com/OsmosysSoftware/document-service/pulls] with a fix.
+
+## Submission Guidelines
+
+### Submitting an Issue
+
+Before you submit an issue, please check the issue tracker to see if a similar issue already exists. The discussions there may provide workarounds.
+
+For us to address and fix a bug, we need to reproduce it. Thus when submitting a bug report, we will ask for a minimal reproduction scenario using a repository or [Gist](https://gist.github.com/). Providing a live, reproducible scenario helps us understand the issue better. Information to include:
+
+- The version of the document-service you are using.
+- Any third-party libraries and their versions.
+- A use-case that demonstrates the issue.
+
+Without a minimal reproduction, we may need to close the issue due to insufficient information.
+
+You can file new issues using our [new issue form](https://github.com/OsmosysSoftware/document-service/issues/new).
+
+### Submitting a Pull Request (PR)
+
+Before submitting a Pull Request (PR), please follow these guidelines:
+
+1. Search GitHub [Pull Requests](https://github.com/OsmosysSoftware/document-service/pulls) to ensure there is no open or closed PR 
+   related to your submission.
+2. Fork this repository.
+3. Make your changes in a new Git branch.
+   ```csharp
```

Comment:
This is a command to be ran in CMD. The formatting need not be in csharp. Remove this.

Reply:
Updated